priv/templates/nodetool

```diff
@@ -35,35 +39,22 @@ main(Args) ->
         end,
 
     case RestArgs of
-        ["ping"] ->
-            %% If we got this far, the node already responsed to a ping, so just dump
-            %% a "pong"
-            io:format("pong\n");
-        ["stop"] ->
-            io:format("~p\n", [rpc:call(TargetNode, init, stop, [], Timeout)]);
-        ["restart"] ->
-            io:format("~p\n", [rpc:call(TargetNode, init, restart, [], Timeout)]);
-        ["reboot"] ->
-            io:format("~p\n", [rpc:call(TargetNode, init, reboot, [], Timeout)]);
-        ["rpc", Module, Function | RpcArgs] ->
-            case rpc:call(TargetNode, list_to_atom(Module), list_to_atom(Function),
-                          [RpcArgs], Timeout) of
-                ok ->
-                    ok;
-                {badrpc, Reason} ->
-                    io:format("RPC to ~p failed: ~p\n", [TargetNode, Reason]),
-                    halt(1);
-                _ ->
-                    halt(1)
-            end;
-        ["rpcterms", Module, Function | ArgsAsString] ->
-            case rpc:call(TargetNode, list_to_atom(Module), list_to_atom(Function),
-                          consult(lists:flatten(ArgsAsString)), Timeout) of
-                {badrpc, Reason} ->
-                    io:format("RPC to ~p failed: ~p\n", [TargetNode, Reason]),
-                    halt(1);
-                Other ->
-                    io:format("~p\n", [Other])
+        ["rpc", Module, Function | ArgsAsString] ->
+            case consult(lists:flatten(ArgsAsString)) of
+                {error, Reason} ->
+                    io:format("Error parsing arguments `~s`, failed with reason: ~s~n", [ArgsAsString, Reason]);
+                {ok, ParsedArgs} when is_list(ParsedArgs) ->
+                    case rpc:call(TargetNode, list_to_atom(Module),
+                                  list_to_atom(Function), ParsedArgs, Timeout) of
+                        {badrpc, Reason} ->
+                            io:format("RPC to ~p failed: ~p~n", [TargetNode, Reason]),
+                            halt(1);
+                        Other ->
+                            io:format("~p~n", [Other])
+                    end;
+                {ok, ParsedArgs} ->
+                    io:format("Error: args `~p` not a list.~n"
+                              "Example to call `foo:bar(1,2,3)` pass `foo bar \"[1, 2, 3].\"`~n", [ParsedArgs])
```

Comment:
wasn't that `foo bar 1. 2. 3.` ?

Reply:
It was, yes. This is the new form that `erl_call` uses. I updated `nodetool` to support this new format as well.

Reply:
ah sweet